### PR TITLE
feat: support skipping node registration

### PR DESCRIPTION
**Pull request by frezbo**
This PR supports skipping node registration from K8s.

This is an adavnced use case and only needs to be used in special cases.
In this mode Kubelet only runs the static pods.

Fixes: #5847

Operations that will be broken:

- `talosctl cluster create` would eventually timeout since it expects
  nodes to be registered.
- `talosctl health` since it expects nodes to be registered.
- `talosctl upgrade-k8s` since it expects nodes to be registered. Static
  pods can still be updated by editing the machine config..

Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec.go [hunk 1]
```diff
@@ -163,7 +166,7 @@ func (ctrl *KubeletSpecController) Run(ctx context.Context, r controller.Runtime
 			return fmt.Errorf("error merging arguments: %w", err)
 		}
 
-		kubeletConfig, err := NewKubeletConfiguration(cfgSpec.ClusterDNS, cfgSpec.ClusterDomain, cfgSpec.ExtraConfig, cfgSpec.DefaultRuntimeSeccompEnabled)
+		kubeletConfig, err := NewKubeletConfiguration(cfgSpec)
```
Comment (frezbo) [inline internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:169]: do i need to pass as?
```suggestion
		kubeletConfig, err := NewKubeletConfiguration(cfgSpec.DeepCopy())
```

Comment (frezbo) [reply internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:169]: the map gets modified here: https://github.com/siderolabs/talos/pull/6021/files#diff-c1756580a1897c92944f52223cd04cfd6279aabfafc71c7bfcd30a0904e6b807R323

**Review by smira — APPROVED** [review]

File: internal/app/machined/pkg/controllers/k8s/kubelet_spec.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/controllers/k8s/kubelet_spec.go:169]: looks like it's fine? as we don't use `cfgSpec` further, and what we get back as a resource is always a copy

**Pull request merged**